src/ffi.rs

```diff
@@ -87,10 +89,21 @@ pub fn to_py_array(array: ArrayRef, py: Python, pyarrow: &PyModule) -> PyResult<
         array.data_type().clone(),
         true,
     )));
-    let array = Box::new(ffi::export_array_to_c(array));
+    let mut new_arr = array;
+    // TODO(Clark): Fix struct array and fixed-size list array slice FFI upstream in arrow2/pyarrow.
```

Comment:
Should we move this logic into `fix_child_array_slice_offsets`? I'd imagine we may need more "fixer" functions in the future